php-checks/src/test/resources/checks/TooManyFieldsInClassCheck.php

```diff
@@ -29,3 +29,76 @@ class B {
   public $f1;
   public $f2;
 };
+
+class ConstructorPropertyPromotionPublic {     // Noncompliant {{Refactor this class so it has no more than 4 fields, rather than the 5 it currently has.}}
+  public function __construct(
+    public string $f1,
+    public string $f2,
+    public string $f3,
+    public string $f4,
+    public string $f5,
+  ) {}
+};
+
```

Comment:
Extra use case:
```
class ConstructorPropertyPromotionPublic {     // Noncompliant {{Refactor this class so it has no more than 4 fields, rather than the 5 it currently has.}}
  public function __construct(
    PUBLIC string $f1,
    PUBLIC string $f2,
    PUBLIC string $f3,
    PUBLIC string $f4,
    PUBLIC string $f5,
  ) {}
};
```

Reply:
@rudy-regazzoni-sonarsource is still in JCL context. Using uppercase for keywords 🤣